kyuubi-common/src/main/scala/org/apache/kyuubi/service/THttpFrontendService.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.service
+
+import java.util.concurrent.{SynchronousQueue, ThreadPoolExecutor, TimeUnit}
+
+import org.apache.hive.service.rpc.thrift._
+import org.apache.thrift.protocol.TBinaryProtocol
+import org.apache.thrift.server.TServlet
+import org.eclipse.jetty.http.HttpMethod
+import org.eclipse.jetty.io.{Connection, EndPoint}
+import org.eclipse.jetty.security.{ConstraintMapping, ConstraintSecurityHandler}
+import org.eclipse.jetty.server._
+import org.eclipse.jetty.server.handler.gzip.GzipHandler
+import org.eclipse.jetty.servlet.{ServletContextHandler, ServletHolder}
+import org.eclipse.jetty.util.security.Constraint
+import org.eclipse.jetty.util.thread.ExecutorThreadPool
+
+import org.apache.kyuubi.{KyuubiException, Logging}
+import org.apache.kyuubi.config.KyuubiConf
+import org.apache.kyuubi.util.NamedThreadFactory
+
+/**
+ * Apache Thrift based hive service rpc
+ *  1. the server side implementation serves client-server rpc calls
+ *  2. the engine side implementations serve server-engine rpc calls
+ */
+abstract class THttpFrontendService(name: String)
```

Comment:
we don't need an abstract class in the common module, let's create a solid class in server module only